mlir/test/Target/Cpp/subscript.mlir

```diff
@@ -4,7 +4,8 @@
 func.func @load_store(%arg0: !emitc.array<4x8xf32>, %arg1: !emitc.array<3x5xf32>, %arg2: index, %arg3: index) {
   %0 = emitc.subscript %arg0[%arg2, %arg3] : <4x8xf32>, index, index
   %1 = emitc.subscript %arg1[%arg2, %arg3] : <3x5xf32>, index, index
-  emitc.assign %0 : f32 to %1 : f32
+  %2 = emitc.lvalue_to_rvalue %0 : !emitc.lvalue<f32>
```

Comment:
I'd expect this to generate
```
size_t v4 = v1[v2][v3]; // emitc.lvalue_to_rvalue %0
v0[v2][v3] = v4; // emitc.assign %2 : f32 to %1 
```
If we don't materialize the value of `v4`, reasoning about the memory effects sounds pretty complicated.

Reply:
See my other comment. The conversion ops don't get materialized on their own but implicitly on every use of the result. I'm happy to discuss alternatives.